### DISCUSSION TITLE
Beginner docs?

### DISCUSSION THREAD

**Discussion by veganstraightedge** [mruby/mruby#6170] [Ideas]
Hi, long time rubyist, new to mruby 👋🏻

- Is there a good place to understand the difference between `mruby` and `ruby`?

---

- Is mruby a strict subset of Ruby? 
  - If so, is there a documented list of what in `ruby` is not in `mruby`?
  - If not, is there documentation of what (if anything) `mruby` has/does that `ruby` doesn't?

---

- Are their beginner guides/docs? Similar to:
  - https://guides.rubygems.org
  - https://guides.rubyonrails.org

---

My context: I'm planning to convert a Kei truck from internal combustion to an EV. And I want to explore using `mruby` for internal controls. Maybe. 🛻🔋⚡️

**Comment by dearblue** [comment]
Unfortunately, there is no such thing as a tutorial.

For information on compiling and building, please see [doc/guides/compile.md](https://github.com/mruby/mruby/blob/master/doc/guides/compile.md).
For other documentation, please check [doc/guides](https://github.com/mruby/mruby/tree/master/doc/guides).

For the differences between mruby and CRuby, [doc/limitations.md](https://github.com/mruby/mruby/blob/master/doc/limitations.md) may be helpful.

If mruby is still too large for your embedded device, please try <https://github.com/mrubyc/mrubyc>.

**Comment by rubyFeedback** [comment]
I think documentation in ruby has always been somewhat of a problem.  Perhaps
we can collectively maintain documentation in ruby in a more relaxed way, that
is allowing for more contributors to add content (solely for documentation, NOT
for features). This may need some kind of wiki, I suppose, or a wiki-like infrastructure,
and probably also needs some kind of co-approval. Perhaps github can be of
help here, but I think the first step that may be useful is to get the ruby core team
to understand that documentation should become a LOT better in general. Ruby
works around this by being an extremely elegant programming language, but
documentation-wise I think there is a lot of improvement possible still.

In regards to some of the questions: to me it always felt as if mruby primarily
targeted experienced C developers already, that is people who know C very
well, and also ruby. I think this is a different focus compared to MRI, which 
targets more people. Ideally one day we could merge everything together to
not have these differences - would be kind of cool if we could have ruby be a
bit like busybox and lua or the linux kernel, picking what we want to use and
see and omitting the rest.

**Comment by matz** [comment]
I am not the right person to write the beginner docs. I know everything about mruby (and Ruby), but I don't know what beginners want to know. If someone comes up with the index, I can fill the contents. Would someone volunteer?

**Reply by jslabovitz** [reply]
I would like to help. I'm not a beginner with CRuby (having used it for 20 years), but I'm new to mruby, and am having a hard time understanding how to get started.
